src/repart/repart.c

```diff
@@ -9262,14 +9269,15 @@ static int run(int argc, char *argv[]) {
                 return r;
         context->from_scratch = r > 0; /* Starting from scratch */
 
-        if (arg_can_factory_reset) {
-                r = context_can_factory_reset(context);
+        r = context_can_factory_reset(context);
+        if (r < 0)
+                return r;
+        if (arg_can_factory_reset)
+                return r == 0 ? EXIT_FAILURE : 0;
+        if (arg_stamp_factory_reset && r > 0) {
+                r = touch("/run/systemd/factory-reset-supported");
```

Comment:
i am really not convinced this should be touch file, at all. thta#s a shitty, horrible api, and means we always have to determine this ahead of time, but that seems wrong to me.

so my thinking was that we can add an env var that overrides things eventually. that way people can write a generator if they like that generates a drop-in for the varlink service that sets the env var, in either direction. and it probably should also mask factory-reset.target or so, to block any attempts to start it early on.